File: 58.最后一个单词的长度.py
#
# @lc app=leetcode.cn id=58 lang=python3
#
# [58] 最后一个单词的长度
#

# @lc code=start
class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        _startFlag = False
        _len = 0
        for c in s[::-1]:
            if c != ' ':
                _startFlag = True
            if _startFlag:
                if c == ' ':
                    break
                else:
                    _len += 1
            
        return _len
        
# @lc code=end

File: 58.最后一个单词的长度.cpp
/*
 * @lc app=leetcode.cn id=58 lang=cpp
 *
 * [58] 最后一个单词的长度
 */

// @lc code=start
class Solution
{
public:
    int lengthOfLastWord(string s)
    {
        string::reverse_iterator it;
        bool start = false;
        int len = 0;
        for (it = s.rbegin(); it != s.rend(); it++)
        {
            if (!start && isalpha(*it))
                start = true;
            if (start)
            {
                if (isalpha(*it))
                    len++;
                else
                    break;
            }
        }
        return len;
    }
};
// @lc code=end
